### DISCUSSION TITLE
Merging two datasets using `xarray.combine_nested` goes wrong

### DISCUSSION THREAD

**Discussion by HJA24** [pydata/xarray#8554] [Q&A]
### What is your issue?

Hi,  I want to merge two `xarray.Dataset`-objects.  I am using version _2023.5.0_.

 My code looks as follows:

```
import numpy as np
import xarray as xr

N_CHAINS = 4
N_DRAWS = 1000
N_PLAYERS = 5

player_idx = [1, 1, 2, 3, 4, 4, 0, 0, 2, 2]
opponent_idx = [0, 3, 1, 4, 1, 1, 1, 4, 3, 3]

h2h_idx = pd.MultiIndex.from_tuples(
    tuple(zip(player_idx, opponent_idx)), names=('player_id', 'opponent_id')
)

obs = xr.Dataset(
    data_vars=dict(
        n_points_won=(['h2h_id'], np.array([11, 11, 8, 9, 4, 11, 7, 11, 11, 11])),
        n_points_lost=(['h2h_id'], np.array([9, 9, 11, 11, 11, 1, 11, 2, 3, 6])),
    ),
    coords=dict(
        h2h_id=(['h2h_id'], h2h_idx),
    )
)

alpha = np.random.rand(N_CHAINS, N_DRAWS, N_PLAYERS, N_PLAYERS) * 100
beta = np.random.rand(N_CHAINS, N_DRAWS, N_PLAYERS, N_PLAYERS) * 100

pos = xr.Dataset(
    data_vars=dict(
        alpha=(['chain', 'draw', 'player_id', 'opponent_id'], alpha),
        beta=(['chain', 'draw', 'player_id', 'opponent_id'], beta),
    ),
    coords=dict(
        chain=(['chain'], list(range(N_CHAINS))),
        draw=(['draw'], list(range(N_DRAWS))),
        player_id=(['player_id'], list(range(N_PLAYERS))),
        oppponent_id=(['opponent_id'], list(range(N_PLAYERS))),
    ),
)
 

combined = xr.combine_nested([obs, pos], concat_dim=['player_id', 'opponent_id']) # this results in an error
```

I am getting the following error:

> ValueError: concat_dims has length 2 but the datasets passed are nested in a 1-dimensional structure

How can I merge the two datasets? 
The goal is to create a `xarray.Dataset` with coordinates `h2h_id`, `player_id`, `opponent_id`, `chain` and `draw`. Basically for each data variable in `obs` I want to concat the corresponding `alpha`- and `beta` values from `pos` of that particular observation (through `player_id` and `opponent_id`)

 Thanks in advance

**Comment by welcome** [comment]
Thanks for opening your first issue here at xarray! Be sure to follow the issue template!
If you have an idea for a solution, we would really welcome a Pull Request with proposed changes.
See the [Contributing Guide](https://docs.xarray.dev/en/latest/contributing.html) for more.
It may take us a while to respond here, but we really value your contribution. Contributors like you help make xarray better.
Thank you!

**Comment by max-sixty** [comment]
Hi @HJA24 — this seems like a usage question rather than an issue with the library. I'll convert it to a discussion. You're more likely to get engagement if you can slim down your example into a much smaller MCVE.

**Comment by TomNicholas** [comment]
It would be helpful if you could slim down the example, but I think there's a misunderstanding of  what `xr.concat` (and by extension `xr.combine_nested`) does. `xr.concat` concatenates variables of the same name along some dimension (the dimension may or may not already exist). Here it appears to me that you're hoping that concatenating the two datasets will concatenate e.g. a variable called `n_points_won` with a variable called `alpha`. If instead you want both variables to live in the same dataset we would call that a `merge` operation. 

`xr.combine_nested` can perform both `concat` and `merge` operations, but it will do so one for each "axis" of the input "grid" of datasets you give it. You're currently giving it two datasets, passed as a single list (so a 1D arrangement of xarray objects), but asking it to perform concat twice (by passing a sequence of 2 dimension names to `concat_dim`. This doesn't make sense - you either have to only pass one dimension name, or instead be trying to combine a list-of-list (i.e. a "2D" arrangement of objects).
